rust/agama-network/src/action.rs

```diff
@@ -0,0 +1,18 @@
+use crate::model::{Connection, DeviceType};
+use uuid::Uuid;
+
+/// Networking actions, like adding, updating or removing connections.
+///
+/// These actions are meant to be processed by [crate::system::NetworkSystem], updating the model
+/// and the D-Bus tree as needed.
+#[derive(Debug)]
+pub enum Action {
+    /// Add a new connection with the given name and type.
+    AddConnection(String, DeviceType),
+    /// Update a connection (replacing the old one).
+    UpdateConnection(Connection),
+    /// Remove the connection with the given Uuid.
+    RemoveConnection(Uuid),
```

Comment:
These actions follow an approach similar to Redux actions. They come from the D-Bus API and they are processed by the [NetworkSystem](https://github.com/openSUSE/agama/blob/0d1defaaa29429dff9896d3c803bdcedbd7c0ff8/rust/agama-network/src/system.rs#L75) that acts as the source of truth. They basically include just the relevant data:

* AddConnection: the NetworkSystem is responsible for creating the connection and it only needs a name and a type.
* UpdateConnection: the D-Bus API already holds a copy of the connection, so it just send the updated version.
* RemoveConnection: NetworkSystem just needs the uuid to remove the connection.

As the D-Bus API already holds a copy of the connection (except in the first case), it would not be a problem to send the whole object if you prefer.

Reply:
@imobachgs well, I just think about possible different backend that do not use uuid for connection identification, so seding whole connection is more generic from my POV. ( I expect that for different backend we will probably also need to change params for Add but it can be solvable later with some factory, but we should not add more limitation than necessary ).

Reply:
Actually, the point is that `Connection` represents kind of a generic struct, no matter which backend you use. Each adapter should return a `Connection`.

In an initial version I had a `NmConnection`, but it was simply too complex for our current scenario.

Reply:
ok, so lets keep it as it is. I think PR is already big enough :)